### ISSUE TITLE
Add Design Namespace

### ISSUE THREAD

**Issue by lsuarez5280** [microsoft/xaml-standard#96]
One thing lacking in Xamarin Forms is the ability to specify the expected type of the binding context in the event it is managed by a UI framework or similar. There are code-based workarounds such as using a service locator pattern to store static references to view models and binding them directly in the page, but that model is, to a large extent, dated and consumes more memory than needed in other patterns.

The design namespace, `http://schemas.microsoft.com/expression/blend/2008`, exists in all Silverlight, WPF, and UWP development and it would be nice to bring some amount of that to Xamarin Forms through XAML Standard 1.0. My primary concern in this scenario is enabling IntelliSense and possibly XAMLC compile-time checking for bindings through something similar to the `DataContext` design-time attribute, but anyone is free to make the case for further portions of that namespace below this proposal.

I imagine this may be somewhat contingent on standardization of a `DataContext`/`BindingContext` property in the final spec, which I understand is presently in proposal.

**Comment by mossyblog** [comment]
I like this personally. Take Unity3D the design composition is you start with a null object (gameobject). You then make additive component changes to that gameobject which decorates it with new behaviours to transform it. If you have a need then to make two components talk to one another then you define properties on each of these components that have inbound type affordance.

So then each of these components work to an agreed contract:

```
Foo : NullObject{
	
	Bar myBar {get; set;}
	private void Update() {
		myBar.x = 200;
	}

}

Bar : NullObject{
	Foo myFoo {get; set;
	
	private void Update() {
		myFoo.x = 200;
	}
}

<Foo x:Name="MyFoo" myBar="{MyBar}"/>
<Bar x:Name="MyBar" myFoo="{MyFoo}"/>

```

Basically the need to specifically defined "Context" isn't as important as it is the inline expression(s) that are encapsulated within the "{}" syntax. By defining the contract myFoo property relies on a Bar class/type, it enables the platform which implements at design-time to enforce the relationship of that code (ie spits out a compile error if there is a mismatch).
